fe/model_rabfe.py

```diff
@@ -579,3 +606,326 @@ class RelativeBindingModel(RelativeModel):
+                complex_box0,
+                prefix="complex_conversion_" + suffix,
+                seed=seed,
+            ),
+            solvent_decouple=self.solvent_decouple_model.simulate_futures(
+                ff_params,
+                mol,
+                solvent_x0,
+                solvent_box0,
+                prefix="solvent_decouple_" + suffix,
+                seed=seed,
+            ),
+            solvent_conversion=self.solvent_conversion_model.simulate_futures(
+                ff_params,
+                mol,
+                solvent_x0,
+                solvent_box0,
+                prefix="solvent_conversion_" + suffix,
+                seed=seed,
+            ),
+            mol=mol,
+            blocker=self.blocker,
+            seed=seed,
+        )
+        return futures
+
+    def predict_from_futures(self, futures: RABFESimulationFutures) -> Tuple[float, float]:
+        dG_complex_decouple, dG_complex_decouple_error = self.complex_decouple_model.predict_from_futures(
+            futures.complex_decouple[0],
+            futures.mol,
+            futures.blocker,
+            futures.complex_decouple[1],
+            futures.complex_decouple[2],
+        )
+        dG_complex_conversion, dG_complex_conversion_error = self.complex_conversion_model.predict_from_futures(
+            futures.complex_conversion[0],
+            futures.mol,
+            futures.complex_conversion[1],
+            futures.complex_conversion[2],
+        )
+        dG_solvent_decouple, dG_solvent_decouple_error = self.solvent_decouple_model.predict_from_futures(
+            futures.solvent_decouple[0],
+            futures.mol,
+            futures.solvent_decouple[1],
+            futures.solvent_decouple[2],
+        )
+        dG_solvent_conversion, dG_solvent_conversion_error = self.solvent_conversion_model.predict_from_futures(
+            futures.solvent_conversion[0],
+            futures.mol,
+            futures.solvent_conversion[1],
+            futures.solvent_conversion[2],
+        )
+
+        rabfe_result = RABFEResult(
+            mol_name=futures.mol.GetProp("_Name"),
+            dG_complex_conversion=dG_complex_conversion,
+            dG_complex_decouple=dG_complex_decouple,
+            dG_solvent_conversion=dG_solvent_conversion,
+            dG_solvent_decouple=dG_solvent_decouple,
+        )
+        rabfe_result.log()
+        dG_err = np.sqrt(
+            dG_complex_conversion_error ** 2
+            + dG_complex_decouple_error ** 2
+            + dG_solvent_conversion_error ** 2
+            + dG_solvent_decouple_error ** 2
+        )
+
+        return rabfe_result.dG_bind, dG_err
+
+    def predict(self, ff_params, mol, core_idxs, suffix) -> Tuple[float, float]:
+        futures = self.simulate_futures(ff_params, mol, core_idxs, suffix)
+        results = self.predict_from_futures(futures)
+        return results
+
+    def set_complex_conversion_schedule(self, schedule: List[int]):
```

Comment:
All these setters could be done with `__getattr__` magic, but as a first pass just expanded them out.

Reply:
Actually, could this be a `dataclass`? I'm not sure I'm reading it correctly yet, but it seems like there's not a lot of internal mutable state?

Reply:
This could definitely be a dataclass